### PR TITLE
Refactor test-utils feature flag

### PR DESCRIPTION
**Pull request by RolandSherwin**
Fix for #1118

Objective:<s>
1. Remove `test-utils` feature flag such that</s>  `cargo test` can be run standalone.<s>
2. Classify between fns used for tests and for bench/example.</s>

Approach:<s>
Introduce two feature flags, `test` and `utils`, where they are used for tests and bench/examples respectively.</s> The different possible scenarios in the code and the approach taken for each are:
1. **Same Crate:** If a fn is used only for testing within the **same crate**, just use `cfg(test)`
2. **Diff Crate:** I.e., a fn defined in one crate is called in another crate. Can be solved by feature gating the function with `cfg(feature="test-utils")` and then importing the fn as `dev-dependency` with `test` feature enabled. Eg: Inside sn_node's cargo.toml, `sn_interface = { path = "../sn_interface", version = "^0.2.3", features= ["test-utils"] }`
4. **Same Crate, diff folder (test/example/bench):** Feature gate it if possible. **Else no `cfg`**

### CODE REVIEW

File: sn_client/src/lib.rs [hunk 1]
```diff
@@ -156,14 +156,11 @@ where
...
-#[cfg(any(test, feature = "test-utils"))]
 use std::sync::Once;
 
-#[cfg(any(test, feature = "test-utils"))]
 static INIT: Once = Once::new();
 
 // TODO: dedupe this w/ sn_node?
-#[cfg(any(test, feature = "test-utils"))]
 /// Initialise logger for tests, this is run only once, even if called multiple times.
 pub fn init_test_logger() {
```
Comment (davidrusu) [inline sn_client/src/lib.rs:165]: Is this used in non-test code? maybe move it into the `mod tests { ..}` block?

Comment (RolandSherwin) [reply sn_client/src/lib.rs:165]: It is used in examples, so cannot have it inside `cfg(test)`. Also there seems to be 4 definitions of `init_test_logger` in the code, maybe shall I have a single one inside `sn_interface` under `feature="test"` and import it in other places?

Comment (joshuef) [reply sn_client/src/lib.rs:165]: > maybe shall I have a single one inside sn_interface under feature="test" and import it in other places?

That sounds good. But then it's probably just `"init_logger"` perhaps?

**Comment by joshuef** [comment]
@RolandSherwin why do we need `cfg(feature="test")` vs something like `cfg(feature="utils")`? 

What are those functions doing? (I just want to be sure they are actually testing code... ie faking some situation, vs just utility code)

**Comment by joshuef** [comment]
> Ah! So for example, functions like init_test_logger() should be under utils while fns like create_test_client_with() should be under test?

As per a HO just now, let's leave off renaming/classifying exports and keep this PR tighter to just `run crate tests with no feature flags` as the aim.

Then later on we can check that everything we're exporting makes sense and classify them a bit better 👍

**Review by davidrusu — APPROVED** [review]

**Pull request merged**